src/enclave/enclave_oe.c

```diff
@@ -61,40 +178,71 @@ void sgxlkl_ethread_init(void)
     sched_tcb->schedctx = (struct schedctx*)((char*)tls_page + tls_offset);
 
     /* Wait until libc has been initialized */
-    while (__libc_state != libc_initialized)
+    while (sgxlkl_enclave_state.libc_state != libc_initialized)
     {
         a_spin();
     }
 
     /* Initialization completed, now run the scheduler */
     __init_tls();
-    _lthread_sched_init(sgxlkl_enclave->stacksize);
+    _lthread_sched_init(sgxlkl_enclave_state.config->stacksize);
     lthread_run();
 
     return;
 }
 
-int sgxlkl_enclave_init(const sgxlkl_config_t* config_on_host)
+static int _read_eeid_config()
 {
-    SGXLKL_ASSERT(config_on_host);
+    const oe_eeid_t* eeid = (oe_eeid_t*)__oe_get_eeid();
+    const char* config_json = (const char*)eeid->data;
+    sgxlkl_enclave_state.libc_state = libc_not_started;
 
+    sgxlkl_enclave_config_t* cfg = malloc(sizeof(sgxlkl_enclave_config_t));
```

Comment:
```suggestion
    sgxlkl_enclave_config_t* cfg = oe_malloc(sizeof(sgxlkl_enclave_config_t));
```

Reply:
That's done by `#include <enclave/oe_compat.h>`

Reply:
I'd prefer it if we only depended on that for things in src/shared.  Anything in src/enclave should only ever be compiled in in-enclave mode and so shouldn't depend on the compatibility header - it makes it easier for someone reading the code to understand what's going on.

Reply:
Yeah, that makes sense, I'll clean that up.

Reply:
(Done)